lib/util/pattern-match.ts

```diff
@@ -0,0 +1,39 @@
+import { minimatch } from './minimatch';
+import { configRegexPredicate, isConfigRegex } from './regex';
+
+export function matchBlobOrRegex(input: string, matcher: string): boolean {
+  // Check if pattern is regex
+  if (isConfigRegex(matcher)) {
+    const autodiscoveryPred = configRegexPredicate(matcher);
+    if (!autodiscoveryPred) {
+      throw new Error(`Failed to parse regex pattern "${matcher}"`);
+    }
+    return autodiscoveryPred(input);
+  }
+  // else use minimatch
+  return minimatch(matcher, { nocase: true }).match(input);
```

Comment:
we should use a cache for Minimatch objects for perf

Reply:
I think it does already thanks to https://github.com/renovatebot/renovate/blob/feat/blob-or-regex/lib/util/minimatch.ts